### ISSUE TITLE
Response slows down.

### ISSUE THREAD

**Issue by castalta** [facebookresearch/ParlAI#4320]
When I constantly send a message to Blender  every second, 
Blender's response gradually slows down, is there any clue to the solution?
(GPU and CPU are getting slower as well.)

I am, but I'm still worried about it. I'm not sure if I should be.
Response Time:  1.8117168741300702
I am, but I'm still worried about it. I'm not sure if I should be.
Response Time:  2.2525314036756754
I will probably just relax and enjoy life.  I have been working hard for a long time.
Response Time:  3.6234819181263447
I was a nerd in high school. I was in the gifted program. I got to go to the best school in the state.
Response Time:  6.251100503839552
I am so excited to begin my new job! I am going to be a manager!
Response Time:  10.890985251404345
I am, but I'm still worried about it. I'm not sure if I should be.
Response Time:  12.854479511268437
I will probably just relax and enjoy life.  I have been working hard for a long time.
Response Time:  12.93592538125813
I will probably just relax and enjoy life.  I have been working hard for a long time.
Response Time:  14.416801642626524
I am so excited to begin my new job! I am going to be a manager!
Response Time:  15.609604943543673
I am so excited to begin my new job! I am going to be a manager!
Response Time:  16.97494798153639
I was a nerd in high school. I was in the gifted program. I got to go to the best school in the state.
Response Time:  20.744939354248345
I will probably just relax and enjoy life.  I have been working hard for a long time.
Response Time:  17.838872809894383
I will probably just relax and enjoy life.  I have been working hard for a long time.
Response Time:  18.126924985088408
I am, but I'm still worried about it. I'm not sure if I should be.
Response Time:  18.43722940608859
I am so excited to begin my new job! I am going to be a manager!
Response Time:  18.673521210439503
I was a nerd in high school. I was in the gifted program. I got to go to the best school in the state.
Response Time:  23.50841996446252

**Comment by klshuster** [comment]
As dialogue history grows, the model is conditioning on more and more encoded context, which requires more computation on blender's part. If you are using context beam blocking, this is especially more computationally expensive

**Comment by castalta** [comment]
Thank you.
I will experiment with the values ​​of beam_context_block and history_size.

**Comment by Jsiewierski11** [comment]
I've experienced this issue as well. I've noticed that more and more VRAM was getting allocated, starting at 6 GBs then growing from there. Did you also see the same relative amount getting allocated to VRAM? Seems like a lot to be getting allocated. Is there maybe anyway to stop the encoded context from growing or making it computationally less expensive? When trying to use it in production it quickly becomes too slow to be used.
